### PR TITLE
Run3-gex90X Clean up a few cfi's for HCAL and SimDB geometry

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Clean up a few cfi's for HCAL and SimDB geometry

#### PR validation:

Use the runTheMatrix test workflow

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by srimanob** [comment]
+Upgrade

Technical PR to clean up the configuration.

File: Configuration/Geometry/python/GeometryDDDSimDB_cff.py [hunk 1]
```diff
@@ -8,8 +8,6 @@
 from Geometry.MuonNumbering.muonNumberingInitialization_cfi import *
 from Geometry.TrackerNumberingBuilder.trackerNumberingGeometryDB_cfi import *
 from Geometry.EcalCommonData.ecalSimulationParameters_cff   import *
-from Geometry.HcalCommonData.hcalSimulationParameters_cff   import *
-from Geometry.HcalCommonData.hcalDDDSimConstants_cfi        import *
-from Geometry.HcalCommonData.hcalSimulationConstants_cfi    import *
-from Geometry.HcalCommonData.caloSimulationParameters_cff   import *
+from Geometry.HcalCommonData.hcalSimDBConstants_cff         import *
 from Geometry.MuonNumbering.muonGeometryConstants_cff       import *
+from Geometry.MuonNumbering.muonOffsetESProducer_cff        import *
```
Comment (cvuosalo) [inline Configuration/Geometry/python/GeometryDDDSimDB_cff.py:13]: Why is the `muonOffsetESProducer_cff` being added? The DDD workflows ran successfully without it, so why is it needed?

Comment (bsunanda) [reply Configuration/Geometry/python/GeometryDDDSimDB_cff.py:13]: It is strictly not needed - I just wanted to symmetrize between ddd and dd4hep parts. It also does not cause harm.

Comment (cvuosalo) [reply Configuration/Geometry/python/GeometryDDDSimDB_cff.py:13]: Please add a comment to the file saying that `muonOffsetESProducer_cff` is not needed but is included for consistency with the DD4hep version.

Comment (bsunanda) [reply Configuration/Geometry/python/GeometryDDDSimDB_cff.py:17]: Added

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**